### PR TITLE
Add integration tests; rename workflows

### PR DESCRIPTION
**Pull request by bonicim**
* Add integration tests 
* Rename workflows 
* Update README
* Remove running tests from precommit config file

### CODE REVIEW

**Comment by bonicim** [comment]
> @bonicim: All the test runs seem to be erroring out. Can you check?

The tests is erroring out because of a network connection; when the server was trying to install zope-interface dependency. 

FYI, I've ran the tests locally and the tests pass.

**Comment by bonicim** [comment]
@shwethanidd Based on the last two runs, the integration tests will fail for different reasons: sometimes the Github server can't [install poetry](https://github.com/eclipse-volttron/volttron-lib-fake-driver/actions/runs/3463493588/jobs/5783807023) and then will error out; and sometimes, the Github server can't install a dependency because of a connection error. Oddly enough, I don't see these errors when running the tests on my personal Github account.

**Comment by bonicim** [comment]
@shwethanidd I got the tests to finally pass on Github's free servers. It appears that I need to install the Fake Driver at least twice (once in the fixture and once again in the test itself) so that my tests can call driver methods such as `scrape_all`, `get_point`, and `set_point`. I don't know why the Github servers require this odd test setup, but I'm now able to successfully run these integration tests on Github.

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -28,8 +28,8 @@ jobs:
 
     strategy:
       matrix:
-        os: [ubuntu-20.04, ubuntu-22.04]
-        python: ['3.8', '3.9', '3.10']
+        os: [ubuntu-22.04]
```
Comment (craig8) [inline .github/workflows/run-tests.yml:31]: Please put the matrix back in.

Comment (bonicim) [reply .github/workflows/run-tests.yml:31]: Done.

**Review by shwethanidd — APPROVED** [review]

**Review by shwethanidd — APPROVED** [review]

**Pull request merged**